lib/general_repo.rb

```diff
@@ -17,7 +18,7 @@ def find_by_id(id)
 
   def create(general_data)
     general_data[:id] ||= (@repository.last.id.to_i + 1).to_s
-    general = General.new(general_data)
+    general = Object.const_get(@class_nm).new(general_data)
```

Comment:
Nice. Pretty much exactly what we talked about. It'l be good to get this implemented in a refactor of my code.